DatadogInternal/Sources/NetworkInstrumentation/URLSession/URLSessionTaskDelegateSwizzler.swift

```diff
@@ -0,0 +1,115 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// Swizzles `URLSessionTaskDelegate` callbacks.
+internal class URLSessionTaskDelegateSwizzler {
```

Comment:
These implementations share a lot of similar code. Would it make sense to create some decoration or even inheritance?

Reply:
I prefer this way to keep them isolated and testable, have had hard time to make them this simpler to read. 

Let's keep this way.